keras_cv/models/__internal__/darknet_utils.py

```diff
@@ -0,0 +1,398 @@
+
+    Args:
+        filters: Integer, the dimensionality of the output spaces (i.e. the number of
+            output filters in used the blocks).
+        hidden_filters: Integer, the dimensionality of the intermediate bottleneck space
+            (i.e. the number of output filters in the bottleneck convolution). If None,
+            it will be equal to filters. Defaults to None.
+        kernel_sizes: A list or tuple representing all the pool sizes used for the
+            pooling layers. Defaults to (5, 9, 13).
+        activation: Activation for the conv layers. Defaults to "silu".
+        name: the prefix for the layer names used in the block.
+
+    Returns:
+        a function that takes an input Tensor representing an SPPBottleneck.
+    """
+    if name is None:
+        name = f"spp{backend.get_uid('spp')}"
+
+    if hidden_filters is None:
+        hidden_filters = filters
+
+    def apply(x):
+        x = DarknetConvBlock(
+            hidden_filters,
+            kernel_size=1,
+            strides=1,
+            activation=activation,
+            name=f"{name}_conv1",
+        )(x)
+        x = [x]
+
+        for kernel_size in kernel_sizes:
+            x.append(
+                layers.MaxPooling2D(
+                    kernel_size,
+                    strides=1,
+                    padding="same",
+                    name=f"{name}_maxpool_{kernel_size}",
+                )(x[0])
+            )
+
+        x = layers.Concatenate(name=f"{name}_concat")(x)
+        x = DarknetConvBlock(
+            filters,
+            kernel_size=1,
+            strides=1,
+            activation=activation,
+            name=f"{name}_conv2",
+        )(x)
+
+        return x
+
+    return apply
+
+
+@tf.keras.utils.register_keras_serializable(package="keras_cv")
+class DarknetConvBlockDepthwise(layers.Layer):
+    """The depthwise conv block used in CSPDarknet.
+
+    Args:
+        filters: Integer, the dimensionality of the output space (i.e. the number of
+            output filters in the final convolution).
+        kernel_size: An integer or tuple/list of 2 integers, specifying the height
+            and width of the 2D convolution window. Can be a single integer to specify
+            the same value both dimensions.
+        strides: An integer or tuple/list of 2 integers, specifying the strides of
+            the convolution along the height and width. Can be a single integer to
+            the same value both dimensions.
+        activation: the activation applied after the final layer. One of "silu",
+            "relu" or "leaky_relu". Defaults to "silu".
+    """
+
+    def __init__(self, filters, kernel_size, strides, activation="silu", **kwargs):
+        super().__init__(**kwargs)
+
+        self.filters = filters
+        self.kernel_size = kernel_size
+        self.strides = strides
+        self.activation = activation
+
+        self.dw_conv = layers.DepthwiseConv2D(
```

Comment:
looks good, but same question as the above: would a sequential model for this actually be more idiomatic?

Reply:
please update to sequential models